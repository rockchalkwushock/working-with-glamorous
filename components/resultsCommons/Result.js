import glamorous from 'glamorous'

const StyledResult = glamorous.li({
  color: 'green'
})

const StyledLink = glamorous.a({
  color: 'orange',
  ':visited': {
    color: 'orange'
  },
  ':hover': {
    color: 'green'
  }
})

StyledLink.displayName = 'StyledLink'
StyledResult.displayName = 'StyledResult'

const Result = ({ link, title }) =>
  <StyledResult>
    <StyledLink href={link}>
      {title}
    </StyledLink>
  </StyledResult>

export default Result
