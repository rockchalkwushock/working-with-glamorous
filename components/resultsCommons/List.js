import glamorous from 'glamorous'

import Result from './Result'

const StyledList = glamorous.ul({
  lineHeight: '1.5',
  listStyle: 'none',
  margin: '0',
  padding: '0',
  paddingBottom: '2em'
})

StyledList.displayName = 'StyledList'

const List = ({ questions }) =>
  <StyledList>
    {questions.map(q =>
      <Result key={Math.random()} link={q.link} title={q.title} />
    )}
  </StyledList>

export default List
