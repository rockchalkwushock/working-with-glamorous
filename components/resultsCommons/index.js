import List from './List'
import Result from './Result'

export { List, Result }
