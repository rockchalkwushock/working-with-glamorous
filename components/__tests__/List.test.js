import { shallow, render, mount } from 'enzyme'

import { List } from '../resultsCommons'

const data = [
  { link: 'a', title: 'a' },
  { link: 'b', title: 'b' },
  { link: 'c', title: 'c' }
]

describe('List Component', () => {
  let ui
  test('should render to the DOM', () => {
    ui = <List questions={data} />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
})
