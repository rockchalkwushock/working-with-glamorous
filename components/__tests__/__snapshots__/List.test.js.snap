// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-8z6uph,
[data-css-8z6uph] {
  line-height: 1.5;
  list-style: none;
  margin: 0px;
  padding: 0px;
  padding-bottom: 2em;
}

.css-1xzrot3,
[data-css-1xzrot3] {
  color: green;
}

.css-v5siu3,
[data-css-v5siu3] {
  color: orange;
}

.css-v5siu3:visited,
[data-css-v5siu3]:visited {
  color: orange;
}

.css-v5siu3:hover,
[data-css-v5siu3]:hover {
  color: green;
}

<List
  questions={
    Array [
      Object {
        "link": "a",
        "title": "a",
      },
      Object {
        "link": "b",
        "title": "b",
      },
      Object {
        "link": "c",
        "title": "c",
      },
    ]
  }
>
  <StyledList>
    <ul
      className="css-8z6uph"
    >
      <Result
        link="a"
        title="a"
      >
        <StyledResult>
          <li
            className="css-1xzrot3"
          >
            <StyledLink
              href="a"
            >
              <a
                className="css-v5siu3"
                href="a"
              >
                a
              </a>
            </StyledLink>
          </li>
        </StyledResult>
      </Result>
      <Result
        link="b"
        title="b"
      >
        <StyledResult>
          <li
            className="css-1xzrot3"
          >
            <StyledLink
              href="b"
            >
              <a
                className="css-v5siu3"
                href="b"
              >
                b
              </a>
            </StyledLink>
          </li>
        </StyledResult>
      </Result>
      <Result
        link="c"
        title="c"
      >
        <StyledResult>
          <li
            className="css-1xzrot3"
          >
            <StyledLink
              href="c"
            >
              <a
                className="css-v5siu3"
                href="c"
              >
                c
              </a>
            </StyledLink>
          </li>
        </StyledResult>
      </Result>
    </ul>
  </StyledList>
</List>
`;

exports[`enzyme.render 1`] = `
.css-8z6uph,
[data-css-8z6uph] {
  line-height: 1.5;
  list-style: none;
  margin: 0px;
  padding: 0px;
  padding-bottom: 2em;
}

.css-1xzrot3,
[data-css-1xzrot3] {
  color: green;
}

.css-v5siu3,
[data-css-v5siu3] {
  color: orange;
}

.css-v5siu3:visited,
[data-css-v5siu3]:visited {
  color: orange;
}

.css-v5siu3:hover,
[data-css-v5siu3]:hover {
  color: green;
}

<ul
  class="css-8z6uph"
>
  <li
    class="css-1xzrot3"
  >
    <a
      class="css-v5siu3"
      href="a"
    >
      a
    </a>
  </li>
  <li
    class="css-1xzrot3"
  >
    <a
      class="css-v5siu3"
      href="b"
    >
      b
    </a>
  </li>
  <li
    class="css-1xzrot3"
  >
    <a
      class="css-v5siu3"
      href="c"
    >
      c
    </a>
  </li>
</ul>
`;

exports[`enzyme.shallow 1`] = `
<StyledList>
  <Result
    link="a"
    title="a"
  />
  <Result
    link="b"
    title="b"
  />
  <Result
    link="c"
    title="c"
  />
</StyledList>
`;
